--- klibc-0.193/kinit/do_mounts.c.orig	2004-12-12 21:12:27.000000000 +0100
+++ klibc-0.193/kinit/do_mounts.c	2004-12-12 21:13:41.211035224 +0100
@@ -1,6 +1,7 @@
 #include <ctype.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <sys/dirent.h>
 #include <sys/mount.h>
 #include <sys/stat.h>
 #include <stdio.h>
@@ -162,8 +163,8 @@
 
 	if (!p)
 		return -1;
-	for (s = p; s < p + size; s += ((struct linux_dirent64 *)s)->d_reclen) {
-		struct linux_dirent64 *d = (struct linux_dirent64 *)s;
+	for (s = p; s < p + size; s += ((struct dirent *)s)->d_reclen) {
+		struct dirent *d = (struct dirent *)s;
 		if (strlen(d->d_name) + 2 > rest)
 			continue;
 		switch (d->d_type) {
