diff -urN klibc-1.1.16.org/klcc/Kbuild klibc-1.1.16/klcc/Kbuild
--- klibc-1.1.16.org/klcc/Kbuild	2006-01-06 07:11:43.000000000 +0100
+++ klibc-1.1.16/klcc/Kbuild	2006-01-06 19:31:36.327836000 +0100
@@ -22,10 +22,10 @@
 	$(Q)echo 'EMAIN=$(KLIBCEMAIN)' >> $@
 	$(Q)echo 'BITSIZE=$(KLIBCBITSIZE)' >> $@
 	$(Q)echo 'VERSION=$(shell cat $(srctree)/usr/klibc/version)' >> $@
-	$(Q)echo 'prefix=$(INSTALLDIR)' >> $@
-	$(Q)echo 'bindir=$(INSTALLDIR)/$(KCROSS)bin' >> $@
-	$(Q)echo 'libdir=$(INSTALLDIR)/$(KCROSS)lib' >> $@
-	$(Q)echo 'includedir=$(INSTALLDIR)/$(KCROSS)include' >> $@
+	$(Q)echo 'prefix=$(rpm_prefix)' >> $@
+	$(Q)echo 'bindir=$(rpm_bindir)' >> $@
+	$(Q)echo 'libdir=$(rpm_libdir)' >> $@
+	$(Q)echo 'includedir=$(rpm_includedir)' >> $@
 
 
 # Generate klcc
--- klibc-2.0.9/klcc/klcc.in.orig	2021-04-29 16:03:19.000000000 +0200
+++ klibc-2.0.9/klcc/klcc.in	2021-05-08 18:55:05.039418288 +0200
@@ -3,22 +3,22 @@
 use IPC::Open3;
 
 # Standard includes
-@includes = ("-I${prefix}/${KCROSS}include/arch/${ARCHDIR}",
-	     "-I${prefix}/${KCROSS}include/bits${BITSIZE}",
-	     "-I${prefix}/${KCROSS}include");
+@includes = ("-idirafter", "${includedir}/arch/${ARCH}",
+	     "-idirafter", "${includedir}/bits${BITSIZE}",
+	     "-idirafter", "${includedir}");
 
 # Default optimization options (for compiles without -g)
 @optopt =  @OPTFLAGS;
 @goptopt = ('-O');
 
 # Standard library directories
-@stdlibpath = ("-L${prefix}/${KCROSS}lib");
+@stdlibpath = ("-L${libdir}");
 
 # Options and libraries to pass to ld; shared versus static
-@staticopt = ("${prefix}/${KCROSS}lib/crt0.o");
-@staticlib = ("${prefix}/${KCROSS}lib/libc.a");
-@sharedopt = (@EMAIN, map { "${prefix}/${KCROSS}lib/$_" } @CRTSHARED);
-@sharedlib = ('-R', "${prefix}/${KCROSS}lib/libc.so");
+@staticopt = ("${libdir}/klibc/crt0.o");
+@staticlib = ("${libdir}/klibc/libc.a");
+@sharedopt = (@EMAIN, map { "${libdir}/klibc/$_" } @CRTSHARED);
+@sharedlib = ('-R', "${libdir}/klibc/libc.so");
 
 # Returns the language (-x option string) for a specific extension.
 sub filename2lang($) {
@@ -113,7 +113,7 @@
 
 $save_temps = 0;		# The -save-temps option
 $verbose = 0;			# The -v option
-$shared = 0;	   		# Are we compiling shared?
+$shared = 1;	   		# Are we compiling shared?
 $debugging = 0;	   		# -g or -p option present?
 $strip = 0;			# -s option present?
 undef $output;			# -o option present?
@@ -186,6 +186,7 @@
     } elsif ( $a eq '-shared' ) {
 	$shared = 1;
     } elsif ( $a eq '-static' ) {
+	push(@ldopt, $a);
 	$shared = 0;
     } elsif ( $a eq '-s' ) {
 	$strip = 1;
diff -urN klibc-1.1.16.org/Makefile klibc-1.1.16/Makefile
--- klibc-1.1.16.org/Makefile	2006-01-06 07:11:43.000000000 +0100
+++ klibc-1.1.16/Makefile	2006-01-06 19:32:24.054818750 +0100
@@ -21,11 +21,11 @@
 export PERL       := perl
 
 # Location for installation
-export prefix      = /usr
-export bindir      = $(prefix)/bin
-export libdir      = $(prefix)/lib
-export mandir      = $(prefix)/man
-export INSTALLDIR  = $(prefix)/lib/klibc
+export prefix      = $(rpm_prefix)
+export bindir      = $(rpm_bindir)
+export libdir      = $(rpm_libdir)
+export mandir      = $(rpm_mandir)
+export INSTALLDIR  = $(rpm_libdir)/klibc
 export INSTALLROOT =
 
 # Create a fake .config as present in the kernel tree
